src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -925,377 +1062,247 @@ TimeState TimeSynchronizationServer::UpdateDSTOffsetState()
-    MutableCharSpan dntp(buffer);
-    err = TimeSynchronizationServer::Instance().GetDefaultNtp(dntp);
-    if (err == CHIP_NO_ERROR)
-    {
-        err = aEncoder.Encode(CharSpan(buffer, dntp.size()));
+    case Commands::SetTrustedTimeSource::Id: {
+        Commands::SetTrustedTimeSource::DecodableType data;
+        ReturnErrorOnFailure(data.Decode(input_arguments, handler->GetAccessingFabricIndex()));
+        return HandleSetTrustedTimeSource(handler, request.path, data);
     }
-    else if (err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
-    {
-        err = aEncoder.EncodeNull();
+    case Commands::SetTimeZone::Id: {
+        Commands::SetTimeZone::DecodableType data;
+        ReturnErrorOnFailure(data.Decode(input_arguments));
+        return HandleSetTimeZone(handler, request.path, data);
+    }
+    case Commands::SetDSTOffset::Id: {
+        Commands::SetDSTOffset::DecodableType data;
+        ReturnErrorOnFailure(data.Decode(input_arguments));
+        return HandleSetDSTOffset(handler, request.path, data);
+    }
+    case Commands::SetDefaultNTP::Id: {
+        Commands::SetDefaultNTP::DecodableType data;
+        ReturnErrorOnFailure(data.Decode(input_arguments));
+        return HandleSetDefaultNTP(handler, request.path, data);
+    }
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedCommand;
     }
-    return err;
-}
-
-CHIP_ERROR TimeSynchronizationAttrAccess::ReadTimeZone(EndpointId endpoint, AttributeValueEncoder & aEncoder)
-{
-    CHIP_ERROR err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
-        const auto & tzList = TimeSynchronizationServer::Instance().GetTimeZone();
-        for (const auto & tzStore : tzList)
-        {
-            ReturnErrorOnFailure(encoder.Encode(tzStore.timeZone));
-        }
-
-        return CHIP_NO_ERROR;
-    });
-
-    return err;
 }
 
-CHIP_ERROR TimeSynchronizationAttrAccess::ReadDSTOffset(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+CHIP_ERROR TimeSynchronizationCluster::AcceptedCommands(const ConcreteClusterPath & path,
+                                                        ReadOnlyBufferBuilder<DataModel::AcceptedCommandEntry> & builder)
 {
-    CHIP_ERROR err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
-        const auto & dstList = TimeSynchronizationServer::Instance().GetDSTOffset();
-        for (const auto & dstOffset : dstList)
-        {
-            ReturnErrorOnFailure(encoder.Encode(dstOffset));
-        }
-
-        return CHIP_NO_ERROR;
-    });
-
-    return err;
+    static constexpr DataModel::AcceptedCommandEntry kAcceptedCommands[] = {
+        Commands::SetUTCTime::kMetadataEntry,    Commands::SetTrustedTimeSource::kMetadataEntry,
+        Commands::SetTimeZone::kMetadataEntry,   Commands::SetDSTOffset::kMetadataEntry,
+        Commands::SetDefaultNTP::kMetadataEntry,
+    };
+    return builder.ReferenceExisting(kAcceptedCommands);
```

Comment:
This seems off: commands have 1 mdandatory and the others are feature map controlled.

You need to respec the feature map and return the right values in both accepted and generated commands.